source/NPC.cpp

```diff
@@ -114,6 +114,8 @@ void NPC::Load(const DataNode &node)
 				for(const DataNode &grand : child)
 					if(grand.Token(0) == "actions" && grand.Size() >= 2)
 						actions[ships.back().get()] = grand.Value(1);
+				if(child.Size() > 2 && ships.back()->Name().length() < 1)
+					ships.back()->SetName(child.Token(2));
```

Comment:
The "name" tag can be used to define the ship's name. If a third token appears in the "ship" line, that will be interpreted as a variant model name, not the ship's name, which could mess things up. (Also, why would you use `length() < 1` instead of `empty()?`)

Reply:
Must've been sleepy :grin: 
This check was due to the inherited behavior from NPC::Load, where the name of the ship is the 3rd entry. I'll add a warning message for 3rd token + child objects (and since #2902 merged, make the changes to those missions to use this featureset as well)